src/libraries/Microsoft.Extensions.Options/tests/SourceGenerationTests/Baselines/NetCoreApp/Validators.g.cs

```diff
@@ -17,11 +17,19 @@ internal sealed partial class __ThirdModelNoNamespaceValidator__
         var baseName = (string.IsNullOrEmpty(name) ? "ThirdModelNoNamespace" : name) + ".";
         var builder = new global::Microsoft.Extensions.Options.ValidateOptionsResultBuilder();
         var context = new global::System.ComponentModel.DataAnnotations.ValidationContext(options);
+        var validationResults = new global::System.Collections.Generic.List<global::System.ComponentModel.DataAnnotations.ValidationResult>();
+        var validationAttributes = new global::System.Collections.Generic.List<global::System.ComponentModel.DataAnnotations.ValidationAttribute>();
 
         context.MemberName = "P5";
         context.DisplayName = baseName + "P5";
-        builder.AddResult(global::__OptionValidationStaticInstances.__Attributes.A1.GetValidationResult(options.P5, context));
-        builder.AddResult(global::__OptionValidationStaticInstances.__Attributes.A2.GetValidationResult(options.P5, context));
+        validationAttributes.Add(global::__OptionValidationStaticInstances.__Attributes.A1);
+        validationAttributes.Add(global::__OptionValidationStaticInstances.__Attributes.A2);
+        if (!global::System.ComponentModel.DataAnnotations.Validator.TryValidateValue(options.P5!, context, validationResults, validationAttributes))
+        {
+            builder.AddResults(validationResults);
+        }
+        validationResults.Clear();
```

Comment:
Why burn cycles clearing the lists?

Reply:
We have cases that we can emit more than one object which we'll use the allocated lists more than once. Look at the same file line 820 as an example.

Reply:
It's worth optimizing the tail case so as not to do the clear operation when there's nothing else happening in the method.

Reply:
I have applied this optimization. Also, I allocate the list now with the initial capacity.